tests/ignite/metrics/test_accumulation.py

```diff
@@ -198,105 +198,123 @@ def compute_mean_std(engine, batch):
-        v = Average(device=device)
-        v.compute()
+        mean_var = Average(device=metric_device)
+        y_true = torch.rand(100, dtype=torch.float64) + torch.randint(0, 10, size=(100,)).double()
+        y_true = y_true.to(device)
 
-    mean_var = Average(device=device)
-    y_true = torch.rand(100, dtype=torch.float64) + torch.randint(0, 10, size=(100,)).double()
-    y_true = y_true.to(device)
+        for y in y_true:
+            mean_var.update(y)
 
-    for y in y_true:
-        mean_var.update(y)
+        m = mean_var.compute()
 
-    m = mean_var.compute()
+        y_true = idist.all_reduce(y_true)
+        assert m.item() == pytest.approx(y_true.mean().item() / idist.get_world_size())
 
-    y_true = idist.all_reduce(y_true)
-    assert m.item() == pytest.approx(y_true.mean().item() / idist.get_world_size())
+        mean_var = Average(device=metric_device)
+        y_true = torch.rand(100, 10, dtype=torch.float64) + torch.randint(0, 10, size=(100, 10)).double()
+        y_true = y_true.to(device)
 
-    mean_var = Average(device=device)
-    y_true = torch.rand(100, 10, dtype=torch.float64) + torch.randint(0, 10, size=(100, 10)).double()
-    y_true = y_true.to(device)
+        for y in y_true:
+            mean_var.update(y)
 
-    for y in y_true:
-        mean_var.update(y)
+        m = mean_var.compute()
 
-    m = mean_var.compute()
+        y_true = idist.all_reduce(y_true)
+        np.testing.assert_almost_equal(
+            m.cpu().numpy(), y_true.mean(dim=0).cpu().numpy() / idist.get_world_size(), decimal=5
+        )
 
-    y_true = idist.all_reduce(y_true)
-    np.testing.assert_almost_equal(
-        m.cpu().numpy(), y_true.mean(dim=0).cpu().numpy() / idist.get_world_size(), decimal=5
-    )
+    # check multiple random inputs as random exact occurencies are rare
+    for _ in range(3):
+        _test("cpu")
+        if device.type != "xla":
+            _test(idist.device())
 
 
 def _test_distrib_geom_average(device):
+    def _test(metric_device):
+        with pytest.raises(NotComputableError):
+            v = GeometricAverage(device=metric_device)
+            v.compute()
 
-    with pytest.raises(NotComputableError):
-        v = GeometricAverage(device=device)
-        v.compute()
+        mean_var = GeometricAverage(device=metric_device)
+        y_true = torch.rand(100, dtype=torch.float64) + torch.randint(0, 10, size=(100,)).double()
+        y_true = y_true.to(device)
 
-    mean_var = GeometricAverage(device=device)
-    y_true = torch.rand(100, dtype=torch.float64) + torch.randint(0, 10, size=(100,)).double()
-    y_true = y_true.to(device)
+        for y in y_true:
+            mean_var.update(y)
 
-    for y in y_true:
-        mean_var.update(y)
+        m = mean_var.compute()
+        log_y_true = torch.log(y_true)
+        log_y_true = idist.all_reduce(log_y_true)
+        assert m.item() == pytest.approx(torch.exp(log_y_true.mean(dim=0) / idist.get_world_size()).item())
```

Comment:
We have some precision issue with 8 TPUs and can be fixed
```suggestion
        np.testing.assert_almost_equal(
            m.item(), torch.exp(log_y_true.mean(dim=0) / idist.get_world_size()).item(), decimal=4
        )
```